core/src/main/java/com/linecorp/armeria/client/HttpClient.java

```diff
@@ -31,6 +32,114 @@
 public interface HttpClient extends ClientBuilderParams {
 
     /**
+     * Creates a new HTTP client that connects to the specified {@code uri} using the default
+     * {@link ClientFactory}.
+     *
+     * @param uri the URI of the server endpoint
+     * @param options the {@link ClientOptionValue}s
+     *
+     * @throws IllegalArgumentException if the scheme of the specified {@code uri} is not an HTTP scheme
+     */
+    static HttpClient of(String uri, ClientOptionValue<?>... options) {
+        return of(ClientFactory.DEFAULT, uri, options);
+    }
+
+    /**
+     * Creates a new HTTP client that connects to the specified {@code uri} using the default
+     * {@link ClientFactory}.
+     *
+     * @param uri the URI of the server endpoint
+     * @param options the {@link ClientOptionValue}s
+     *
+     * @throws IllegalArgumentException if the scheme of the specified {@code uri} is not an HTTP scheme
+     */
+    static HttpClient of(String uri, ClientOptions options) {
+        return of(ClientFactory.DEFAULT, uri, options);
+    }
+
+    /**
+     * Creates a new HTTP client that connects to the specified {@code uri} using an alternative
+     * {@link ClientFactory}.
+     *
+     * @param factory an alternative {@link ClientFactory}
+     * @param uri the URI of the server endpoint
+     * @param options the {@link ClientOptionValue}s
+     *
+     * @throws IllegalArgumentException if the scheme of the specified {@code uri} is not an HTTP scheme
+     */
+    static HttpClient of(ClientFactory factory, String uri, ClientOptionValue<?>... options) {
+        return new HttpClientBuilder(uri).factory(factory).options(options).build();
+    }
+
+    /**
+     * Creates a new HTTP client that connects to the specified {@code uri} using an alternative
+     * {@link ClientFactory}.
+     *
+     * @param factory an alternative {@link ClientFactory}
+     * @param uri the URI of the server endpoint
+     * @param options the {@link ClientOptionValue}s
+     *
+     * @throws IllegalArgumentException if the scheme of the specified {@code uri} is not an HTTP scheme
+     */
+    static HttpClient of(ClientFactory factory, String uri, ClientOptions options) {
```

Comment:
What do you think about having versions that take sessionoption instead of ClientFactory?